### ISSUE TITLE
[🐛 Bug]: browser.getPuppeteer() doesn't work in cucumber hook when run UI tests using "selenium-standalone" service v. 8.6.8

### ISSUE THREAD

**Issue by Yurii397** [webdriverio/webdriverio#10079]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.6.8

### Node.js Version

v16.17.1

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
capabilities: [
    {
      maxInstances: 1,
      browserName: 'chrome', 
      'goog:chromeOptions': {
        args: ['window-size=1920,1080', '--disable-gpu', '--no-sandbox'],
        prefs: {
          directory_upgrade: true,
          prompt_for_download: false,
          'download.default_directory': global.downloadFolder,
        },
      },
    },
  ],
```

### What happened?

I'm doing migration of our test automation framework which uses the following dependencies:

-     "@wdio/cli": "^8.6.8",
-     "@wdio/cucumber-framework": "^8.6.8",
-     "@wdio/local-runner": "^8.6.8",
-     "@wdio/selenium-standalone-service": "^8.6.8",
-     "@wdio/spec-reporter": "^8.6.8",
-     "@wdio/types": "8.6.8", 
    
In cucumber hook. ie. in "beforeFeature"  we are getting puppeteer instances like puppeteer = await browser.getPuppeteer()

but after upgrading version of @wdio/selenium-standalone-service from "^6.12.1"  to "^8.6.8" it stopped work, 

I wrap it into try catch and get this one 
    
<img width="1262" alt="image" src="https://user-images.githubusercontent.com/46851143/228296566-e31c94ed-cd63-45fb-9152-ce8c600ce6b6.png">

### What is your expected behavior?

start getting puppeteer instance as it was in previous versions

### How to reproduce the bug.

I thought something can be wrong with our TAF and setup new project from scratch using the tutorial - https://webdriver.io/docs/gettingstarted

The issue was reproduced again. 

<img width="1490" alt="image" src="https://user-images.githubusercontent.com/46851143/228299095-e7c1008b-474d-4cf3-8b34-47ab705f5be3.png">

### Relevant log output

```typescript
% npx wdio run ./wdio.conf.ts

Execution of 1 workers started at 2023-03-28T16:02:59.753Z

2023-03-28T16:02:59.947Z INFO @wdio/cli:launcher: Run onPrepare hook
2023-03-28T16:03:03.441Z INFO @wdio/cli:launcher: Run onWorkerStart hook
2023-03-28T16:03:03.442Z INFO @wdio/local-runner: Start worker 0-0 with arg: run,./wdio.conf.ts
[0-0] 2023-03-28T16:03:04.346Z INFO @wdio/local-runner: Run worker command: run
[0-0] RUNNING in chrome - file:///features/login/login.feature
[0-0] 2023-03-28T16:03:05.089Z INFO webdriver: Initiate new session using the WebDriver protocol
[0-0] 2023-03-28T16:03:05.170Z INFO webdriver: [POST] http://localhost:4444/wd/hub/session
[0-0] 2023-03-28T16:03:05.170Z INFO webdriver: DATA {
[0-0]   capabilities: {
[0-0]     alwaysMatch: { browserName: 'chrome', acceptInsecureCerts: true },
[0-0]     firstMatch: [ {} ]
[0-0]   },
[0-0]   desiredCapabilities: { browserName: 'chrome', acceptInsecureCerts: true }
[0-0] }
[0-0] puppeteerBrowser
[0-0] 2023-03-28T16:03:19.277Z ERROR @wdio/utils:shim: socket hang up
[0-0] 2023-03-28T16:03:19.291Z INFO webdriver: COMMAND navigateTo("https://the-internet.herokuapp.com/login")
[0-0] 2023-03-28T16:03:19.292Z INFO webdriver: [POST] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a/url
[0-0] 2023-03-28T16:03:19.292Z INFO webdriver: DATA { url: 'https://the-internet.herokuapp.com/login' }
[0-0] 2023-03-28T16:03:24.410Z INFO webdriver: RESULT null
[0-0] 2023-03-28T16:03:24.413Z INFO webdriver: COMMAND findElement("css selector", "#username")
[0-0] 2023-03-28T16:03:24.413Z INFO webdriver: [POST] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a/element
[0-0] 2023-03-28T16:03:24.414Z INFO webdriver: DATA { using: 'css selector', value: '#username' }
[0-0] 2023-03-28T16:03:24.439Z INFO webdriver: RESULT {
[0-0]   'element-6066-11e4-a52e-4f735466cecf': 'a9b59a7d-155b-4c3d-a26b-4941e91818a8'
[0-0] }
[0-0] 2023-03-28T16:03:24.444Z INFO webdriver: COMMAND elementClear("a9b59a7d-155b-4c3d-a26b-4941e91818a8")
[0-0] 2023-03-28T16:03:24.445Z INFO webdriver: [POST] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a/element/a9b59a7d-155b-4c3d-a26b-4941e91818a8/clear
[0-0] 2023-03-28T16:03:24.478Z INFO webdriver: RESULT null
[0-0] 2023-03-28T16:03:24.478Z INFO webdriver: COMMAND elementSendKeys("a9b59a7d-155b-4c3d-a26b-4941e91818a8", "<Screenshot[base64]>")
[0-0] 2023-03-28T16:03:24.478Z INFO webdriver: [POST] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a/element/a9b59a7d-155b-4c3d-a26b-4941e91818a8/value
[0-0] 2023-03-28T16:03:24.478Z INFO webdriver: DATA { text: 'tomsmith' }
[0-0] 2023-03-28T16:03:24.626Z INFO webdriver: RESULT null
[0-0] 2023-03-28T16:03:24.627Z INFO webdriver: COMMAND findElement("css selector", "#password")
[0-0] 2023-03-28T16:03:24.627Z INFO webdriver: [POST] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a/element
[0-0] 2023-03-28T16:03:24.627Z INFO webdriver: DATA { using: 'css selector', value: '#password' }
[0-0] 2023-03-28T16:03:24.642Z INFO webdriver: RESULT {
[0-0]   'element-6066-11e4-a52e-4f735466cecf': '011b82ea-d713-4299-a9ca-6832691931b3'
[0-0] }
[0-0] 2023-03-28T16:03:24.645Z INFO webdriver: COMMAND elementClear("011b82ea-d713-4299-a9ca-6832691931b3")
[0-0] 2023-03-28T16:03:24.645Z INFO webdriver: [POST] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a/element/011b82ea-d713-4299-a9ca-6832691931b3/clear
[0-0] 2023-03-28T16:03:24.676Z INFO webdriver: RESULT null
[0-0] 2023-03-28T16:03:24.676Z INFO webdriver: COMMAND elementSendKeys("011b82ea-d713-4299-a9ca-6832691931b3", "SuperSecretPassword!")
[0-0] 2023-03-28T16:03:24.677Z INFO webdriver: [POST] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a/element/011b82ea-d713-4299-a9ca-6832691931b3/value
[0-0] 2023-03-28T16:03:24.677Z INFO webdriver: DATA { text: 'SuperSecretPassword!' }
[0-0] 2023-03-28T16:03:24.743Z INFO webdriver: RESULT null
[0-0] 2023-03-28T16:03:24.744Z INFO webdriver: COMMAND findElement("css selector", "button[type="submit"]")
[0-0] 2023-03-28T16:03:24.744Z INFO webdriver: [POST] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a/element
[0-0] 2023-03-28T16:03:24.744Z INFO webdriver: DATA { using: 'css selector', value: 'button[type="submit"]' }
[0-0] 2023-03-28T16:03:24.760Z INFO webdriver: RESULT {
[0-0]   'element-6066-11e4-a52e-4f735466cecf': '05c9e900-cc8c-49ef-94ee-0f978f9f17c7'
[0-0] }
[0-0] 2023-03-28T16:03:24.763Z INFO webdriver: COMMAND elementClick("05c9e900-cc8c-49ef-94ee-0f978f9f17c7")
[0-0] 2023-03-28T16:03:24.764Z INFO webdriver: [POST] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a/element/05c9e900-cc8c-49ef-94ee-0f978f9f17c7/click
[0-0] 2023-03-28T16:03:25.125Z INFO webdriver: RESULT null
[0-0] 2023-03-28T16:03:25.128Z INFO webdriver: COMMAND findElement("css selector", "#flash")
[0-0] 2023-03-28T16:03:25.129Z INFO webdriver: [POST] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a/element
[0-0] 2023-03-28T16:03:25.129Z INFO webdriver: DATA { using: 'css selector', value: '#flash' }
[0-0] 2023-03-28T16:03:25.146Z INFO webdriver: RESULT {
[0-0]   'element-6066-11e4-a52e-4f735466cecf': 'c8eece74-e9d9-420c-923b-36568c7eae59'
[0-0] }
[0-0] 2023-03-28T16:03:25.149Z INFO webdriver: COMMAND findElements("css selector", "#flash")
[0-0] 2023-03-28T16:03:25.149Z INFO webdriver: [POST] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a/elements
[0-0] 2023-03-28T16:03:25.149Z INFO webdriver: DATA { using: 'css selector', value: '#flash' }
[0-0] 2023-03-28T16:03:25.167Z INFO webdriver: RESULT [
[0-0]   {
[0-0]     'element-6066-11e4-a52e-4f735466cecf': 'c8eece74-e9d9-420c-923b-36568c7eae59'
[0-0]   }
[0-0] ]
[0-0] 2023-03-28T16:03:25.173Z INFO webdriver: COMMAND findElement("css selector", "#flash")
[0-0] 2023-03-28T16:03:25.174Z INFO webdriver: [POST] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a/element
[0-0] 2023-03-28T16:03:25.174Z INFO webdriver: DATA { using: 'css selector', value: '#flash' }
[0-0] 2023-03-28T16:03:25.190Z INFO webdriver: RESULT {
[0-0]   'element-6066-11e4-a52e-4f735466cecf': 'c8eece74-e9d9-420c-923b-36568c7eae59'
[0-0] }
[0-0] 2023-03-28T16:03:25.192Z INFO webdriver: COMMAND getElementText("c8eece74-e9d9-420c-923b-36568c7eae59")
[0-0] 2023-03-28T16:03:25.192Z INFO webdriver: [GET] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a/element/c8eece74-e9d9-420c-923b-36568c7eae59/text
[0-0] 2023-03-28T16:03:25.211Z INFO webdriver: RESULT You logged into a secure area!
[0-0] ×
[0-0] 2023-03-28T16:03:25.214Z INFO webdriver: COMMAND deleteSession()
[0-0] 2023-03-28T16:03:25.214Z INFO webdriver: [DELETE] http://localhost:4444/wd/hub/session/4848a98ae72737001ab6a426d5aaf14a
[0-0] 2023-03-28T16:03:25.332Z INFO webdriver: RESULT null
[0-0] PASSED in chrome - file:///features/login/login.feature
2023-03-28T16:03:25.451Z INFO @wdio/cli:launcher: Run onWorkerEnd hook
2023-03-28T16:03:25.452Z INFO @wdio/cli:launcher: Run onComplete hook
2023-03-28T16:03:25.452Z INFO @wdio/selenium-standalone-service: shutting down all browsers
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Can you provide a reproducible example? I created a new WebdriverIO project and replaced the generated test with:

```js
import LoginPage from '../pageobjects/login.page.js'
describe('My Login application', () => {
    it('should login with valid credentials', async () => {
        const puppeteer = await browser.getPuppeteer()
        const page = (await puppeteer.pages())[0]
        await page.setRequestInterception(true)
        page.on('request', interceptedRequest => {
            if (interceptedRequest.url().endsWith('webdriverio.png')) {
                return interceptedRequest.continue({
                    url: 'https://user-images.githubusercontent.com/10379601/29446482-04f7036a-841f-11e7-9872-91d1fc2ea683.png'
                })
            }

            interceptedRequest.continue()
        })

        // continue with WebDriver commands
        await browser.url('https://v6.webdriver.io')
        await browser.pause(2000)
    })
})
```

Everything worked as expected.

**Comment by christian-bromann** [comment]
nevermind, it is reproducible

**Comment by diemol** [comment]
This should be fixed on Selenium 4.9.1

**Comment by christian-bromann** [comment]
Awesome, great work @diemol and thanks for reporting back. Closing.
